test/functional/p2p_fingerprint.py

```diff
@@ -101,34 +90,31 @@ def run_test(self):
 
         # Check that getdata request for stale block succeeds
         self.send_block_request(stale_hash, node0)
-        test_function = lambda: self.last_block_equals(stale_hash, node0)
-        self.wait_until(test_function, timeout=3)
+        node0.wait_for_block(stale_hash, timeout=3)
 
         # Check that getheader request for stale block header succeeds
         self.send_header_request(stale_hash, node0)
-        test_function = lambda: self.last_header_equals(stale_hash, node0)
-        self.wait_until(test_function, timeout=3)
+        node0.wait_for_header(hex(stale_hash), timeout=3)
 
         # Longest chain is extended so stale is much older than chain tip
         self.nodes[0].setmocktime(0)
-        tip = self.nodes[0].generatetoaddress(1, self.nodes[0].get_deterministic_priv_key().address)[0]
+        self.nodes[0].generatetoaddress(1, self.nodes[0].get_deterministic_priv_key().address)
```

Comment:
20047: should pick missing changes from 20466 we merged earlier
```suggestion
        block_hash = int(self.nodes[0].generatetoaddress(1, self.nodes[0].get_deterministic_priv_key().address)[-1], 16)
```